### PR TITLE
followup to add http server close method

### LINKED ISSUE #25009: follow-up for Re-enable http test server Close() calls
The original issue: #24546 

https://github.com/kubernetes/kubernetes/pull/24595 gave the first try, However there are still some leftover.

### PR DESCRIPTION
**Pull request by zhouhaibing089**
Fixes #25009, a follow up of https://github.com/kubernetes/kubernetes/pull/24595.

### CODE REVIEW

File: pkg/proxy/userspace/proxier_test.go [hunk 1]
```diff
@@ -124,6 +126,9 @@ func init() {
 	}
 	udpServerPort = int32(udpServerPortValue)
 	go udp.Loop()
+
+	ret := m.Run()
+	os.Exit(ret)
```
Comment (lavalamp) [inline pkg/proxy/userspace/proxier_test.go]: I think the os.Exit will cause the program to exit before it does the deferred tcp.Close().

**Comment by lavalamp** [comment]
Thanks!

**Comment by zhouhaibing089** [comment]
@lavalamp thanks for remind this, yes, the deferred function will not run when `os.Exit` is called. I added one more `tcp.Close()` before `os.Exit(ret)`.

It is safe to call it multiple times.

**Comment by janetkuo** [comment]
LGTM but PR needs a rebase now

**Comment by zhouhaibing089** [comment]
@janetkuo just rebased, and thanks for review.

**Comment by lavalamp** [comment]
ok to test

**Comment by zhouhaibing089** [comment]
The integration test show this error: 

```
goroutine 5465 [semacquire, 4 minutes]:
sync.runtime_Semacquire(0xc82bbd4edc)
	/usr/local/go/src/runtime/sema.go:47 +0x26
sync.(*WaitGroup).Wait(0xc82bbd4ed0)
	/usr/local/go/src/sync/waitgroup.go:127 +0xb4
net/http/httptest.(*Server).Close(0xc82bbd4ea0)
	/usr/local/go/src/net/http/httptest/server.go:189 +0x25d
k8s.io/kubernetes/test/integration.TestQuota(0xc82bcedc20)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/integration/quota_test.go:118 +0x1f40
testing.tRunner(0xc82bcedc20, 0x31b40e8)
	/usr/local/go/src/testing/testing.go:473 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:582 +0x892
```

I think it happened due to there are two go routines running which do not expect the server to be closed, see https://github.com/kubernetes/kubernetes/blob/master/test/integration/quota_test.go#L78 and https://github.com/kubernetes/kubernetes/blob/master/test/integration/quota_test.go#L97.

**Comment by zhouhaibing089** [comment]
@janetkuo tests passes, please consider add `ok-to-merge` and `lgtm`, thanks. :)

**Pull request merged**